### PR TITLE
doc: fix indentation, punctuation, and links

### PR DESCRIPTION
**Pull request by Stebalien**
Clippy was complaining about list indentation and paragraph separation, so I fixed that and a few other issues (links to types, etc.).

I may have gone a little ham but... I couldn't stop once I started.

### CODE REVIEW

**Comment by Stebalien** [comment]
You can test this locally with `cargo check --all --all-targets` (and `cargo doc`). That should make sure there are no dangling links, at the very least.

Other than that, you'll have to read it to make sure I didn't make any glaring mistakes.

I found most of the missing periods with:

```bash
rg -t rust -U '^ *///.*[^.:]\n *(///\s*[A-Z\n]|[^/ ])'
```

File: frc53_nft/src/types.rs [hunk 1]
```diff
@@ -7,87 +7,94 @@ use fvm_ipld_encoding::RawBytes;
 use fvm_shared::address::Address;
 use fvm_shared::ActorID;
 
+#[cfg(doc)]
+use super::state::Cursor;
```
Comment (rvagg) [inline frc53_nft/src/types.rs:11]: hah, so we have to import them for "doc" if they're not already imported here in order for the doc links to work?

File: frc53_nft/src/types.rs [hunk 2]
```diff
@@ -97,13 +104,14 @@ impl RecipientData for MintIntermediate {
     }
 }
 
-/// Intermediate data used by transfer_return to construct the return data
+/// Intermediate data used by [`NFT::transfer_return`][`super::NFT::transfer_return`] to construct
```
Comment (rvagg) [inline frc53_nft/src/types.rs:107]: what's going on with both of these links? is this a typo that they're both included?

File: frc53_nft/src/util.rs [hunk 3]
```diff
@@ -28,14 +28,16 @@ impl OperatorSet for BitField {
...
+/// Maintains set-like invariants in-memory by maintaining sorted order of the underlying array.
+///
+/// Insertion and deletion are O(n) operations but we expect operator lists to be a relatively small
+/// size.
+// TODO: benchmark this against some other options such as
+// - BTreeSet in memory, Vec serialized
+// - BTreeSet in memory and serialization
+// - HashSets...
+// - Hamt<ActorID, ()>
+// - Amt<ActorID>
```
Comment (rvagg) [inline frc53_nft/src/util.rs:40]: this is going to work OK? `//` after the `///`? should you just stuck the TODO bit inside the `impl` maybe?

**Review by rvagg — APPROVED** [review]
are the `.`'s actually necessary to satisfy clippy or are we just being pedantic?

File: frc53_nft/src/types.rs (hunk 1 above)
Comment (Stebalien) [reply frc53_nft/src/types.rs:11]: Yep: https://doc.rust-lang.org/rustdoc/advanced-features.html#cfgdoc-documenting-platform-specific-or-feature-specific-information

Well, I could also link to it with the full path, but then all the links become rather long.

File: frc53_nft/src/types.rs (hunk 2 above)
Comment (Stebalien) [reply frc53_nft/src/types.rs:107]: It's markdown formatting (reference style link). This is the explicit linking I was avoiding in https://github.com/filecoin-project/actors-utils/pull/237#discussion_r1704844288.

Rustdoc has some interesting handling of links: https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html#linking-to-items-by-name

(I guess the backticks around the reference aren't necessary, but they work)

File: frc53_nft/src/util.rs (hunk 3 above)
Comment (Stebalien) [reply frc53_nft/src/util.rs:40]: I can move it inside, I just didn't want it in the doc comment.

**Comment by Stebalien** [comment]
> are the .'s actually necessary to satisfy clippy or are we just being pedantic?

No, I was just being pedantic. I like to ocasionally browse rustdocs for our projects and this:

```rust
/// This is some text
/// This is some other text
```

Will get formatted as:

> This is some text This is some other text

The only thing I _needed_ here was to fix:

```rust
/// - some very long bulletpoint item.
/// continuation...
```

Which needs to be:

```rust
/// - some very long bulletpoint item.
///   continuation...
```

File: frc53_nft/src/util.rs (hunk 3 above)
Comment (Stebalien) [reply frc53_nft/src/util.rs]: I moved it above to make it clear.

**Pull request merged**